api/src/main/java/org/apache/iceberg/ContentScanTask.java

```diff
@@ -66,7 +66,8 @@
 
   @Override
   default long estimatedRowsCount() {
-    double scannedFileFraction = ((double) length()) / file().fileSizeInBytes();
+    long splitOffset = (file().splitOffsets() != null) ? file().splitOffsets().get(0) : 0L;
```

Comment:
could `splitOffsets` be an empty list?

Reply:
That is a good question, and I believe the answer is no. The javadoc for `ContentFile#splitOffsets()` says
"Returns list of recommended split locations, if applicable, null otherwise." Thus if there are no split offsets, null rather than an empty list should be returned. The javadoc also says "The returned list must be sorted in ascending order."